pkg/transform/json/json_test.go

```diff
@@ -76,18 +109,9 @@ func TestTransform(t *testing.T) {
 	if val, ok := transformed.Fields["ts"]; !ok || val != "not a time" {
 		t.Errorf("Expected field 'ts' to be 'not a time', but got '%s'", val)
 	}
+}
 
-	// Parse a log field that isn't json
-	record = &types.Record{
-		Fields: map[string]interface{}{
-			"log":   "this totally isn't json",
-			"other": "foobar",
-		},
-	}
-	transformed, err = Transform(record)
-	if err != nil {
-		t.Fatal(err)
-	}
+func checkLogFieldThatIsNotJSON(t *testing.T, transformed *types.Record) {
```

Comment:
First time im seeing this pattern, but i really like it.
 are there any blog posts about it that we can use to help evangelize it?

Reply:
I'm not aware of a blog post (although this is relevant but not the same https://blog.golang.org/subtests). This is simply dividing a function (a test) into multiple smaller functions. The nice thing is you can pass `t *testing.T` so you can treat the smaller funcs as tests. :)